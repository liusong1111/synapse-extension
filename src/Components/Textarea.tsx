import * as React from "react";

interface AppProps {
  onChange: any;
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return (
      <div>
        <textarea onChange={this.props.onChange} />
      </div>
    )
  }
}