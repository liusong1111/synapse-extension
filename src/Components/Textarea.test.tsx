import * as React from 'react'
import { shallow } from 'enzyme'
import Textarea from './Textarea'

describe('Textarea', () => {
  let wrapper
  beforeEach(() => wrapper = shallow(<Textarea onChange={jest.fn()} />))

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render Textarea', () => {
    expect(wrapper.find('textarea').length).toEqual(1)
  });

  it('should render value', () => {

  });


  // it('should render title text', () => {
  //   wrapper.setProps({ title: 'Import Mnemonic'})
  //   expect(wrapper.text()).toEqual('Import Mnemonic')
  // });


})