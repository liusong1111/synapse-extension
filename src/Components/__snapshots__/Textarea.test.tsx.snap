// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render correctly 1`] = `
<div>
  <textarea
    onChange={[MockFunction]}
  />
</div>
`;
